### PR TITLE
Feature/labels and textarea

### PR DESCRIPTION
**Pull request by rav-pradhan**
### What

- Added styles for textarea and label components.
- Added documentation for textarea and first pass on implementing 'code example' component inspired from DST/GDS systems. This component has syntax highlighting and uses a JS helper function to extract html from example `njk` files.
- Updated nav component to differentiate between Styles and Components
- Added accessible tabs JS for the code example component. This is currently taken from the w3 docs just for quickness and ease; we could look at rolling our own tabs component when we get to implementing that in the design system

image

### How to review

Ensure styles and code changes make sense. Load the docs locally and review content etc

### Who can review

Anyone

### CODE REVIEW

File: src/views/colours.njk [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Colours
+tags: styles
```
Comment (rav-pradhan) [inline src/views/colours.njk:3]: Do let me know if there's a better way of distinguishing between nav categories

File: src/scss/main.scss [hunk 2]
```diff
@@ -16,3 +16,9 @@
 'base/typography',
 'base/utilities/hide',
 'base/generic';
+
+// Components
+@import
+'components/input',
+'components/label',
+'components/example'
```
Comment (rav-pradhan) [inline src/scss/main.scss:24]: Currently loading in the example component with the main design system styles. Would be good to bounce ideas on how best to separate out the docs-specific styles from the main design system without introducing too much duplication (could be work that's covered in a later ticket)

File: src/views/components/textarea/index.njk [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+<header class="mb-8">
+    <div class="text-grey-2 text-lg mb-1">Components</div>
+    <h1 class="mb-6">{{title}}</h1>
+    <p>{{description}}</p>
+</header>
+
+{% from "example.njk" import example %}
+{{ example({group: "components", item: "textarea", example: "default", html: true, open: false, size: "m"}) }}
+
+<section class="my-8">
```
Comment (rav-pradhan) [inline src/views/components/textarea/index.njk:20]: In Vue and Svelte etc. there's the concept of slots which would be quite handy here, I think, as these sections will most likely be reused across our docs. Is there any way we could mimic this with nunjucks?

**Review by hamishtaplin — CHANGES_REQUESTED** [review]

File: src/views/colours.njk (hunk 1 above)
Comment (hamishtaplin) [reply src/views/colours.njk:3]: I created a "collection" for styles in this PR

https://github.com/ONSdigital/dp-design-system/pull/9/files#diff-c306e0a99961a16f5c5c83996caa0958b94006d97f97475049ea3a08036bb5b0R13

File: src/scss/main.scss (hunk 2 above)
Comment (hamishtaplin) [reply src/scss/main.scss:24]: Currently, the docs-specific styles are written in Tailwind for this reason.

I think we need to decide wether we are happy to:

1. mix and match tailwind with some Sass
2. refactor now (or very soon) to move the tailwind styles into Sass
3. stick to tailwind only

My preference is 3 (I want to move quickly and tailwind helps that) at the moment but happy to consider the other options.

File: src/views/components/textarea/index.njk (hunk 3 above)
Comment (hamishtaplin) [reply src/views/components/textarea/index.njk:20]: Coming from a React background, this is exactly my problem with njk—you can't compose components together. Unless you can and I haven't figured it out yet but I suspect the syntax would be some horrible workaround using `set` and the params of `macro`.

File: src/scss/abstracts/variables/_colors.scss [hunk 4]
```diff
@@ -52,9 +52,10 @@ $tintable-colors: (
 // Assignment
 $text-color: $black;
 $text-color-inverse: $white;
+$color-focus: $jaffa-orange;
 
 $error-color: $ruby-red;
-
+$link-color: $indigo-blue;
```
[the remark this answers is not in the document]
Comment (armstrongb) [reply src/scss/abstracts/variables/_colors.scss:58]: Add to the [discussion page for colours](https://github.com/ONSdigital/dp-design-system/discussions/14) if this is beyond scope of this PR?

File: src/scss/abstracts/variables/_colors.scss [hunk 5]
```diff
@@ -52,9 +52,10 @@ $tintable-colors: (
 // Assignment
 $text-color: $black;
 $text-color-inverse: $white;
+$color-focus: $jaffa-orange;
```
[the remark this answers is not in the document]
Comment (armstrongb) [reply src/scss/abstracts/variables/_colors.scss:55]: Add to the [discussion page for colours](https://github.com/ONSdigital/dp-design-system/discussions/14) if this is beyond scope of this PR?

File: src/scss/main.scss (hunk 2 above)
Comment (rav-pradhan) [reply src/scss/main.scss:24]: I see now. I agree with approach 3 - I'll refactor the example component's markup to be written with Tailwind

File: src/views/colours.njk (hunk 1 above)
Comment (rav-pradhan) [reply src/views/colours.njk:3]: Thanks. Will implement as such

**Review by hamishtaplin — APPROVED** [review]

**Pull request merged**